code/modules/shuttle/emergency.dm

```diff
@@ -226,24 +235,30 @@
 				mode = SHUTTLE_IDLE
 				timer = 0
 		if(SHUTTLE_CALL)
+			if(time_left <= 135 && crash && !crash_announced)
+				//Notify people that a coder is an absolute ass
+				event_announcement.Announce("An error has been detected in shuttle docking systems - brace for impact!", "Don't drink and drive", new_sound = 'sound/misc/notice1.ogg')
+				crash_announced = TRUE
 			if(time_left <= 0)
 				//move emergency shuttle to station
 				if(dock(shuttle_master.getDock("emergency_home")))
 					setTimer(20)
 					return
+				//Be a dick to everyone and ruin their day (okay, not everyone, only those who are not buckled)
+				if(crash)
+					var/area/target = locate(crash_target)
+					for(var/mob/M in target)
```

Comment:
I'd still make this `for(var/mob/living/M in target)` to automatically cut out ghosts from the loop (saves a small bit of processing and avoids any potential weirdness with trying to call things on them)

Reply:
Oh, right. I am doing this like this so the ghosts can also hear and see the crash. I could exclude them from the `Weaken` proc but I think this is nicer that way. (It does nothing to ghosts.)